services/wallet/router/filter.go

```diff
@@ -0,0 +1,103 @@
+package router
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+func filterRoutesV2(routes [][]*PathV2, amountIn *big.Int, fromLockedAmount map[uint64]*hexutil.Big) [][]*PathV2 {
+	if len(fromLockedAmount) == 0 {
+		return routes
+	}
+
+	routesAfterNetworkCompliance := filterNetworkComplianceV2(routes, fromLockedAmount)
+	return filterCapacityValidationV2(routesAfterNetworkCompliance, amountIn, fromLockedAmount)
+}
+
+// filterNetworkComplianceV2 performs the first level of filtering based on network inclusion/exclusion criteria.
+func filterNetworkComplianceV2(routes [][]*PathV2, fromLockedAmount map[uint64]*hexutil.Big) [][]*PathV2 {
+	var filteredRoutes [][]*PathV2
+	fromIncluded, fromExcluded := setupRouteValidationMapsV2(fromLockedAmount)
+
+	for _, route := range routes {
+		if isValidForNetworkComplianceV2(route, fromIncluded, fromExcluded) {
+			filteredRoutes = append(filteredRoutes, route)
+		}
+	}
+	return filteredRoutes
+}
+
+// isValidForNetworkComplianceV2 checks if a route complies with network inclusion/exclusion criteria.
+func isValidForNetworkComplianceV2(route []*PathV2, fromIncluded, fromExcluded map[uint64]bool) bool {
+	for _, path := range route {
+		if fromExcluded[path.From.ChainID] {
+			return false
+		}
+		if _, ok := fromIncluded[path.From.ChainID]; ok {
+			fromIncluded[path.From.ChainID] = true
+		}
+	}
+
+	for _, included := range fromIncluded {
+		if !included {
+			return false
+		}
+	}
+
+	return true
+}
+
+// setupRouteValidationMapsV2 initializes maps for network inclusion and exclusion based on locked amounts.
+func setupRouteValidationMapsV2(fromLockedAmount map[uint64]*hexutil.Big) (map[uint64]bool, map[uint64]bool) {
+	fromIncluded := make(map[uint64]bool)
+	fromExcluded := make(map[uint64]bool)
+
+	for chainID, amount := range fromLockedAmount {
+		if amount.ToInt().Cmp(zero) == 0 {
```

Comment:
Until we add the check for `fromLockedAmount` validation, we should check `if amount.ToInt().Cmp(zero) <= 0 {`, right?